master/docs/examples/gitlab.cfg

```diff
@@ -0,0 +1,223 @@
+# in one fell swoop, making it easier to have a large number
+# of similar projects.
+addGitLabProject('git@gitlab.example.com:build/gnu-hello', workernames)
+addGitLabProject('git@gitlab.example.com:build/gnu-time', workernames)
+
+####### BUILDBOT SERVICES
+
+# 'services' is a list of BuildbotService items like reporter targets. The
+# status of each build will be pushed to these targets. buildbot/reporters/*.py
+# has a variety to choose from, like IRC bots.
+
+c['services'] = []
+
+####### PROJECT IDENTITY
+
+# the 'title' string will appear at the top of this buildbot installation's
+# home pages (linked to the 'titleURL').
+
+c['title'] = "Gnu Hello GitLab"
+c['titleURL'] = "https://gitlab.example.com/build/"
+
+# the 'buildbotURL' string should point to the location where the buildbot's
+# internal web server is visible. This typically uses the port number set in
+# the 'www' entry below, but with an externally-visible host name which the
+# buildbot cannot figure out without some help.
+
+c['buildbotURL'] = "http://buildbot.example.com:8010/"
+
+# CONFIGME: buildbot authentication
+# This example tries to show nothing to anonymous users.
+# authz = util.Authz(
+#   allowRules=[
+#     util.AnyEndpointMatcher(role="platform"),
+#     util.AnyEndpointMatcher(role="snorglepuss-does-not-exist", defaultDeny=True),
+#   ],
+#   roleMatchers=[
+#     util.RolesFromGroups()
+#   ]
+# )
+
+# minimalistic config to activate new web UI
+c['www'] = dict(
+    port=8010,
+    # CONFIGME: GitLab application
+    # Enter the URL of your buildbot gitlab hook into the GitLab UI at
+    #  "User Settings / Applications / Add New Application"
+    # to obtain the 2nd and 3rd parameters for GitLabAuth.
+    auth=util.GitLabAuth("https://gitlab.example.com",
+                         "036777777777777777777777777777777777777777777777777777777cea803a",
+                         "bae69888888888888888888888888888888888888888888888888888888888ec"),
+    # CONFIGME: buildbot authentication
+    #authz=authz,
+    change_hook_dialects=dict(
+       gitlab={
+           # CONFIGME: GitLab webhook
+           # Create a webhook in the GitLab UI at
+           #  project / Settings / Integrations / Add Webhook 
+           # with a URL of e.g.
+           #   http://buildbot.example.com:8010/change_hook/gitlab
+           # with push and merge request triggers checked, 
+           # and with a secret that matches the one here (please change):
+           'secret': 'changeme',
+       },
+    ),
+    plugins=dict(waterfall_view={}, console_view={}, grid_view={}))
+
+# CONFIGME: GitLab access token
+# Create an access token in the GitLab UI at
+#   "User Settings / Access Tokens / Add a personal access token"
+# to obtain 1st and 2nd parameters for GitLabStatusPush.
+gl = reporters.GitLabStatusPush('Sas23r293232ddzfHAvd', context='youchoose/this', baseURL='https://gitlab.example.com', verbose=True)
+c['services'].append(gl)
+
+c['buildbotNetUsageData'] = None
```

Comment:
I'd prefer this is opt-out so we have a bit of stats from gitlab users.
For gitlab this is especially interresting, as gitlab already has a fairly powerful CI engine builtin.

Reply:
yes please